YANG Packages Update for IETF 121

rgwilton opened the pull request:
Start updating YANG packages for IETF 121:
* Update based on changes to Module versioning and YANG Semver.
* Get modules compiling again.
* Add initial attend for supporting mounted schema.
* Remove "local" flag, instead give naming guidance for the package.
* Remove namespaces.

reshad-equinix commented on yang-packages/draft-ietf-netmod-yang-packages.txt:
```diff
@@ -368,18 +361,25 @@ Internet-Draft                YANG Packages                 October 2022
+       supersedes any 'implemented' revision of the module listed in an
+       included package module list.  The 'replaces-revision' leaf-list
+       is used to indicate which 'implemented' or 'import-only' module
+       revisions are replaces by this module revision.  This allows a
```
Nit s/replaces/replaced/

rgwilton replied:
Fixed

reshad-equinix commented on yang-packages/draft-ietf-netmod-yang-packages.txt:
```diff
@@ -1226,32 +1153,32 @@ Internet-Draft                YANG Packages                 October 2022
 
-     typedef pkg-identifier {
-       type rev:name-revision;
+     typedef version-or-rev-date {
```
Ideally this should be in YANG semver document?

rgwilton replied:
Maybe, although, I'm not sure whether it really matters.  Potentially having it in the YANG Semver would mean that it is in only one place. Thoughts @xorrkaz ?

reshad-equinix commented on yang-packages/draft-ietf-netmod-yang-packages.txt:
```diff
@@ -1361,55 +1276,43 @@ Internet-Draft                YANG Packages                 October 2022
             one of the included packages.";
        }
 
-       leaf local {
```
I recall we discussed this but not sure whether we closed on whether we can do without it.

rgwilton replied:
Yes, I don't think that this is a closed issue.  I was discussing it with Jason, and maybe deleting it is a bit pre-emptive.  I'm sure that there will be some further discussion on whether this is a good flag to have.

reshad-equinix commented on yang-packages/draft-ietf-netmod-yang-packages.txt:
```diff
@@ -1663,33 +1558,105 @@ Internet-Draft                YANG Packages                 October 2022
+
+         leaf "mount-path" {
+           type string;
+           description
```
Do we need a reference to the schema mount RFC?

reshad-equinix reviewed (approved):
Left some comments but these contents can be published. Please change my affiliation.

rgwilton commented on yang-packages/draft-ietf-netmod-yang-packages.txt:
```diff
@@ -5,12 +5,12 @@
 Internet-Draft                                                 R. Rahman
 Intended status: Standards Track                               J. Clarke
-Expires: 28 April 2023                               Cisco Systems, Inc.
+Expires: 18 April 2025                               Cisco Systems, Inc.
```
Fixed

rgwilton commented:
Updates from IETF 122:
- Removed replaced-revision, since we decided that was no longer required.
- Changed mounted-packages to take a list of packages, and changed the semantics to combine rather than replace (which is consistent with how included-packages are merged), but we haven't necessarily reached consensus on this.
- I've not included any text about automatic merging, I started writing some text, but I think it needs some further thought to get right (but, I still think that this would be a good idea if we can achieve it).
- We should also discuss the path format used for the mounted packages, and the resolution rules for more specific paths vs less specific paths (which is arguably impossible in the general case, particularly if regexes for key matches are allowed).
